modules/tracing/mule-tracer-exporter-impl/src/main/java/org/mule/runtime/tracer/exporter/impl/MutableMuleTraceState.java

```diff
@@ -77,13 +83,16 @@ public boolean isEmpty() {
   @Override
   public void forEach(BiConsumer<String, String> biConsumer) {
     // From the remote state we don't export the ancestor.
+    // But if the flag is disable we take it as part of the trace state.
+    // If it shouldn't remove the ancestor span id key value
+    // it takes it as any other span id.
     remoteState.forEach((key, value) -> {
-      if (!key.equals(ANCESTOR_MULE_SPAN_ID)) {
+      if (isRemoveMuleAncestorSpanIdKeyValue || !key.equals(ANCESTOR_MULE_SPAN_ID)) {
```

Comment:
This is a bit confusing because the comment says "if it shouldn't remove the ancestor" and then the code evaluates a boolean called isRemoveMuleAncestorSpanIdKeyValue expecting it to be true. Is the naming correct?